src/AppBundle/Controller/SchoolClassController.php

```diff
@@ -0,0 +1,304 @@
+     */
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    /**
+     * Get all schoolClasses of a school.
+     *
+     * @return SchoolClass[] of a school, every schoolClass is public
+     *
+     * @param int $schoolId id of the school
+     *
+     * @SWG\Parameter(
+     *     name="schoolId",
+     *     in="path",
+     *     type="integer",
+     *     description="id of the school"
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns all schoolsClass, every schoolClass is public",
+     *     @SWG\Schema(
+     *         type="array",
+     *         @SWG\Items(ref="#/definitions/SchoolClass")
+     *     )
+     * )
+     * @Rest\Get("/fromSchool/{schoolId}")
+     */
+    public function getAllFromSchoolAction($schoolId)
+    {
+        $restresult = $this->getDoctrine()->getRepository('AppBundle:SchoolClass')->findAllFromSchool($schoolId);
+        if (null === $restresult) {
+            $restresult = array();
+        }
+
+        return $restresult;
+    }
+
+    /**
+     * Get one schoolClass.
+     *
+     * @param int $classId id of the schoolClass
+     *
+     * @return School[] SchoolClass with id=$id
+     *
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns one schoolClass with id={id}",
+     *     @Model(type=SchoolClass::class)
+     * )
+     * @SWG\Parameter(
+     *     name="id",
+     *     in="path",
+     *     type="integer",
+     *     description="id of the schoolClass"
+     * )
+     * @Rest\Get("/{id}")
+     */
+    public function getOneAction($id)
+    {
+        $singleresult = $this->getDoctrine()->getRepository('AppBundle:SchoolClass')->find($id);
+        if (null === $singleresult) {
+            return new View('SchoolClass not found', Response::HTTP_NOT_FOUND);
+        }
+
+        return $singleresult;
+    }
+
+    /**
+     * Create schoolClass.
+     *
+     * @param Request $request newObjectRequest
+     *
+     * @return SchoolClass SchoolClass just created
+     *
+     * @SWG\Parameter(
+     *     name="name",
+     *     in="formData",
+     *     type="string",
+     *     required=true,
+     *     description="The schoolClass name"
+     * )
+     * @SWG\Parameter(
+     *     name="schoolId",
+     *     in="formData",
+     *     type="integer",
+     *     required=true,
+     *     description="The school"
+     * )
+     * @SWG\Response(
+     *     response=200,
+     *     description="Returns one SchoolClass with id",
+     *     @Model(type=SchoolClass::class)
+     * )
+     * @SWG\Response(
+     *     response=403,
+     *     description="Returns 403 HTTP if the user has no permission",
+     * )
+     * @SWG\Response(
+     *     response=400,
+     *     description="Returns 400 HTTP if values are wrong",
+     * )
+     * @Rest\Post("")
+     */
+    public function newAction(Request $request)
```

Comment:
I think the code is not so equal than it seams to be. Example:
* In the newAction you get the SchoolId from the Request, in the modify and delete action, you get it from the old SchoolClass object, 
* the permssions between DangerPoints and SchoolClasses are different.
* The permission method of the updateAction is split into two parts, so that you will get an permission error even if the SchoolClass does not exists.

Also it is very complicated to implement a gerneral method, if the code is on different branches.

I suggest, that we create a new issue for this.

Reply:
OK. When all the controllers are implemented, then we could analyse them and add an issue if necessary.